cmd/govim/vimstate.go

```diff
@@ -113,7 +118,95 @@ func (v *vimstate) setUserBusy(args ...json.RawMessage) (interface{}, error) {
 	if v.userBusy || v.config.QuickfixAutoDiagnosticsDisable || !v.quickfixIsDiagnostics {
 		return nil, nil
 	}
-	return nil, v.updateQuickfix()
+	return nil, v.redefineDiagnostics()
+}
+
+func (v *vimstate) redefineDiagnostics() error {
```

Comment:
I'd suggest perhaps creating `diagnostics.go` to split this file up a bit? It's become a bit of a grab bag

Reply:
Good point, fixed